gen3-integration-tests/tests/test_gen3_workflows.py

```diff
@@ -504,35 +504,60 @@ def test_nextflow_workflow(self):
                 response["ContentLength"] > 0
             ), f"Expected to have some data in {expected['filename']}. But found empty file instead. Response :{response}"
 
-            # Verify the contents of the command file match the expected commands.
-            command_file_keys = [
-                file["Key"]
-                for file in s3_file_list
-                if file["Key"].endswith("/.command.sh")
+            test_files = [
+                "/.command.sh",
+                "/.command.err",
+                "/.command.out",
+                "/.command.log",
+                "/.exitcode",
             ]
-            assert (
-                len(command_file_keys) == 1
-            ), f"Expected to find exactly one .command.sh file for task '{task_name}', but got {len(command_file_keys)}. Files: {command_file_keys}"
-
-            command_file_key = command_file_keys[0]
 
-            # get contents of .command.sh file
-            response = self.gen3_workflow.get_bucket_object_with_boto3(
-                object_path=f"{self.s3_storage_config.bucket_name}/{command_file_key}",
-                s3_storage_config=self.s3_storage_config,
-                user=self.valid_user,
-            )
-
-            try:
-                command_script_contents = response["Body"].read().decode("utf-8")
-            except Exception as e:
-                raise ValueError(
-                    f"Failed to read or decode content of {command_file_key} from S3. Error: {e}"
+            for test_file_name in test_files:
+                # Verify the contents of the command file match the expected commands.
+                matching_keys = [
+                    file["Key"]
+                    for file in s3_file_list
+                    if file["Key"].endswith(test_file_name)
+                ]
+                assert (
+                    len(matching_keys) == 1
+                ), f"[{task_name}] Expected to find exactly one {test_file_name} file', but got {len(matching_keys)}. Files: {matching_keys}"
+
+                s3_key = matching_keys[0]
+
+                # get contents of file
+                response = self.gen3_workflow.get_bucket_object_with_boto3(
+                    object_path=f"{self.s3_storage_config.bucket_name}/{s3_key}",
+                    s3_storage_config=self.s3_storage_config,
+                    user=self.valid_user,
                 )
 
-            assert (
-                expected["command"] in command_script_contents
-            ), f"Expected to find `{expected['command']}` in the .command.sh file for task {task_name}, but got {command_script_contents}"
+                try:
+                    file_contents = response["Body"].read().decode("utf-8")
+                except Exception as e:
+                    raise ValueError(
+                        f"[{task_name}] Failed to read or decode content of {s3_key} from S3. Error: {e}"
+                    )
```

Comment:
I think this would be better, so we can get the whole stacktrace instead of just the stringified error:
```
except Exception as e:
    logger.error(f"[{task_name}] Failed to read or decode content of {s3_key} from S3. Error: {e}")
    raise
```
There's another block with the same logic in this file, could you fix both